File: CTask/src/org/ShapeSHifter420/Filters/GenderFilter.java
package org.ShapeSHifter420.Filters;

import org.ShapeSHifter420.Pushes.*;

public class GenderFilter implements IFilter {
    public boolean isAllow(IPush push){
        return push instanceof IGenderPush;
    };
    public boolean test(IPush push, ISystem sys){
        IGenderPush genPush = (IGenderPush) push;
        return  (genPush.GetGender().equals(sys.GetGender()));
    }
}

File: CKotlin/Filters/GenderFilter.kt
package Filters

import Pushes.IAgePush
import Pushes.IGenderPush
import Pushes.IPush
import Pushes.ISystemData

class GenderFilter : IFilter {
    override fun allow(push: IPush): Boolean = push is IGenderPush

    override fun filtration(push: IPush,sysData: ISystemData): Boolean {
        val genderPush = push as IGenderPush
        return sysData.gender == genderPush.gender
    }
}